india_forecast_app/adjuster.py

```diff
@@ -0,0 +1,91 @@
+""" Adjuster code, adjust forecast by last 7 days of ME"""
+from datetime import datetime, timedelta
+from typing import Optional
+
+import pandas as pd
+from pvsite_datamodel.sqlmodels import ForecastSQL, ForecastValueSQL, GenerationSQL
+from sqlalchemy import INT, cast, text
+from sqlalchemy.sql import func
+
+# Wad to get all the forecast for the last 7 days made, at this time.
+# And find the ME for each forecast horizon
+
+"""
+Here is the SQL query that it based off
+
+select 
+AVG(generation.generation_power_kw - forecast_values.forecast_power_kw)
+-- generation.generation_power_kw,
+-- forecast_values.forecast_power_kw,
+-- forecast_values.start_utc,
+,horizon_minutes
+-- *
+from forecast_values 
+JOIN forecasts ON forecasts.forecast_uuid = forecast_values.forecast_uuid 
+join generation on generation.start_utc = forecast_values.start_utc
+WHERE forecast_values.start_utc >= '2024-11-05' 
+AND forecasts.site_uuid = 'adaf6be8-4e30-4c98-ac27-964447e9c8e6'
+AND generation.site_uuid = 'adaf6be8-4e30-4c98-ac27-964447e9c8e6'
+and extract(hour from forecasts.created_utc) = 7
+group by horizon_minutes
+-- order by forecast_values.start_utc
+"""
+
+
+def get_me_values(
+    session, hour: int, site_uuid: str, start_datetime: Optional[datetime] = None
+) -> pd.DataFrame:
+    """
+    Get the ME values for the last 7 days for a given hour, for a given hour creation time
+
+    Args:
+    hour: the hour of when the forecast is created
+    site_uuid: the site this is for
+    start_datetime:: the start datetime to filter on.
+    session: sqlalchemy session
+
+    """
+
+    if start_datetime is None:
+        start_datetime = datetime.now() - timedelta(days=7)
+
+    query = session.query(
+        func.avg(ForecastValueSQL.forecast_power_kw - GenerationSQL.generation_power_kw),
+        ForecastValueSQL.horizon_minutes,
+    )
+
+    # join
+    query = query.join(ForecastSQL)
+
+    # round Generation start_utc and join to forecast start_utc
+    start_utc_minute_rounded = (
+        cast(func.date_part("minute", GenerationSQL.start_utc), INT)
+        / 15
+        * text("interval '15 min'")
+    )
+    start_utc_hour = func.date_trunc("hour", GenerationSQL.start_utc)
+    generation_start_utc = start_utc_hour + start_utc_minute_rounded
+    query = query.filter(generation_start_utc == ForecastValueSQL.start_utc)
+
+    # only include the last 7 days
```

Comment:
I know it mentions 7 days here but currently if a `start_datetime` is passed then it can be longer than 7 days? Is it worth adding another check that it isn't longer than 7 days ago or is this intentional for flexibility? If so then maybe just the comment can change

Reply:
yea, thanks